File: BodyFit.Entities/AppUser.cs
using BodyFit.Entities.Enums;
using MongoDB.Bson;
using System.ComponentModel;

namespace BodyFit.Entities
{
    public class AppUser : MongoBaseEntity
    {
        [DisplayName("Kullanıcı")]
        public string Username { get; set; }

        [DisplayName("E-Posta Adresi")]
        public string EmailAddress { get; set; }


        [DisplayName("Şifre")]
        public string Password { get; set; }


        [DisplayName("Kullanıcı Rolü")]
        public ObjectId RoleId { get; set; }

        public int Height { get; set; }
        public decimal TargetWeight { get; set; }

        public ExercisePeriod ExercisePeriod { get; set; }
        public TargetType TargetType { get; set; }
        public DailyActivityType DailyActivityType { get; set; }

        public Dictionary<PreferencesType, object> Preferences { get; set; } = new Dictionary<PreferencesType, object>();
    }

    public enum TargetType
    {
        None,
        LoseWeight,
        Strong,
        Healty
    }

    public enum DailyActivityType
    {
        None,
        Sedentary,
        Dynamic,
    }   

    public enum ExercisePeriod
    {
        None,
        Everyday,
        AFewTimesInWeek,
        AFewTimesInMonth,
        Never
    }
}

File: bodyfit.client/src/app/models/entities/appUser.ts
import { DailyActivityType } from './../enums/DailyActivityType';
import { ExercisePeriod } from './../enums/ExercisePeriod';
import { TargetType } from './../enums/TargetType';
import { BaseEntity } from './baseEntity';
import { GenderType } from './../enums/GenderType';

export class AppUser extends BaseEntity{
    username:string;
    emailAddress:string;
    height:number;
    targetWeight:number;
    exercisePeriod:ExercisePeriod;
    targetType:TargetType;
    dailyActivityType:DailyActivityType;
    age:number;
    gender:GenderType;

    constructor(){
        super();
        this.age = 30;
        this.gender = GenderType.Male;
        this.username = "";
        this.emailAddress = "";
        this.height = 0;
        this.targetWeight = 0;
        this.exercisePeriod = ExercisePeriod.None;
        this.targetType = TargetType.None;
        this.dailyActivityType = DailyActivityType.None;
    }
}